wrangler/src/ctrlAccounts/heal.js

```diff
@@ -1,14 +1,56 @@
-import { response } from 'cfw-easy-utils'
-import { heal } from '../@utils/trust'
+import { response } from 'cfw-easy-utils';
+import { StrKey } from 'stellar-base';
+import { heal } from '../@utils/trust';
 
 export default async ({ request, env }) => {
   const {
-    sourceaccount,
-    oldturret,
-    newturret,
-    functionhash
-  } = await request.json()
-
-  const healresponse = await heal(sourceaccount, oldturret, newturret, functionhash)
-  return response.json(healresponse)
-}
\ No newline at end of file
+    controlAccountId,
+    removeTurretAccountId,
+    newTurretAccountId,
+    txFunctionHash,
+    timestamp,
+    userAccountId,
+    fee,
+  } = await request.json();
+
+  // light input parsing
+  // validate the event timestamp was recent - cap at 5 minutes
+  if (timestamp > Date.now() - 5 * 60 * 1000) {
+    throw {
+      status: 400,
+      message: 'The heal event timestamp was over 5 minutes ago',
+    };
+  }
+
+  // validate public keys are actually public keys
+  if (
+    !StrKey.isValidEd25519PublicKey(controlAccountId) ||
+    !StrKey.isValidEd25519PublicKey(removeTurretAccountId) ||
+    !StrKey.isValidEd25519PublicKey(newTurretAccountId) ||
+    !StrKey.isValidEd25519PublicKey(userAccountId)
+  ) {
+    throw {
+      status: 400,
+      message: 'Please use a valid ed25519 public key for all accountIds.',
+    };
+  }
+
+  // validate the fee is something believable
+  if (fee < 100 || fee > 10000000) {
+    throw {
+      status: 400,
+      message: 'Please enter a valid fee in stroops between 100 and 10,000,000',
+    };
+  }
```

Comment:
can be convinced some input verification is not necessary, but I think it will make interacting with the API easier.

Figured this can be pushed down somewhere once more endpoints make use of this

Reply:
I was using this in the code i never pushed for input validations, so it does work.  I agree it's not "necessary" but it's also not harmful, so i'm of the opinion to keep it.